sampler_test.go

```diff
@@ -86,3 +86,23 @@ func TestRatioSamplerNever(t *testing.T) {
 		Span: apm.SpanID{255, 255, 255, 255, 255, 255, 255, 255},
 	}))
 }
+
+func TestRatioSamplerExtended(t *testing.T) {
+	s := apm.NewRatioSampler(0.5).(apm.ExtendedSampler)
+
+	result := s.SampleExtended(apm.SampleParams{
+		TraceContext: apm.TraceContext{Span: apm.SpanID{255, 0, 0, 0, 0, 0, 0, 0}},
+	})
+	assert.Equal(t, apm.SampleResult{
+		Sampled:    false,
+		SampleRate: 0.5,
```

Comment:
Based on the spec, this result is wrong, right? I was expecting to find the special case https://github.com/elastic/apm-agent-go/pull/804/files#diff-ade132f78e2d3e9afadc8eabd0ac434bR105 inside `ExtendedSample` somehow

Reply:
The special case is handled elsewhere (transaction.go), to keep the logic in one place, rather than in every implementation of ExtendedSampler.

Reply:
Having said that, I think we're missing a test where the sample rate is > 0, sampling decision is false, and es=s:0. I'll add one.